server/modules/storage/git/storage.js

```diff
@@ -207,7 +224,23 @@ module.exports = {
       } else {
         // -> Asset
 
-        if (!fileExists && ((item.before > 0 && item.after === 0) || (item.deletions > 0 && item.insertions === 0))) {
+        if (fileExists && ((item.before === item.after) || (item.deletions === 0 && item.insertions === 0))) {
+          // Asset was renamed by git, so rename in DB
+          WIKI.logger.info(`(STORAGE/GIT) Asset marked as renamed: from ${item.oldPath} to ${item.relPath}`)
+
+          const fileHash = assetHelper.generateHash(item.relPath)
+          const assetToRename = await WIKI.models.assets.query().findOne({ hash: fileHash })
+          if (assetToRename) {
+            await WIKI.models.assets.query().patch({
+              filename: item.relPath,
+              hash: fileHash
+            }).findById(assetToRename.id)
+            await assetToRename.deleteAssetCache()
+          } else {
+            WIKI.logger.info(`(STORAGE/GIT) Asset was not found in the DB, nothing to rename: ${item.relPath}`)
+          }
+          continue
```

Comment:
This should theoretically work for assets, but wasn't accessible in my testing because `diff.files` earlier in the file wasn't being populated when a binary was renamed.

Reply:
This is a limitation of git-js, which doesn't notice binaries being renamed: https://github.com/steveukx/git-js/issues/885